### PR TITLE
Fixed text alignment for RTL selects

### LINKED ISSUE #5684: Bad text alignment for RTL selects
https://forum.getodk.org/t/right-to-left-text-alignment-for-select-one-and-select-multiple-options/42291

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #5684 

#### What has been done to verify that this works as intended?
I've tested the fix manually.

#### Why is this the best possible solution? Were any other approaches considered?
There is nothin to discuss here.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
Please test the fix on many different Android versions. It's important to check those older - Android 5.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/adapters/AbstractSelectListAdapter.java [hunk 1]
```diff
@@ -137,8 +137,6 @@ void setUpButton(TextView button, int index) {
         button.setTextSize(TypedValue.COMPLEX_UNIT_DIP, QuestionFontSizeUtils.getQuestionFontSize());
         button.setText(HtmlUtils.textToHtml(prompt.getSelectChoiceText(filteredItems.get(index))));
         button.setTag(items.indexOf(filteredItems.get(index)));
-        button.setGravity(LocalizedApplicationKt.isLTR(context) ? Gravity.START : Gravity.END);
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/adapters/AbstractSelectListAdapter.java:140]: It looks like we no longer need to do that programmatically.

**Review by seadowg — APPROVED** [review]

**Comment by dbemke** [comment]
In All widgets in selects with RTL applied radio buttons are to the right but choices are too the left. 
Verified on the CI built app on Android 10,13. On the master version the issue doesn't occur.
rtlSelect

**Comment by srujner** [comment]
I've also got few screens to compare behaviors on Android 5.1 and 13

**Android 5.1 at the top, 13 at the bottom**
Screenshot from 2023-07-26 15-34-34

**Comment by grzesiek2010** [comment]
I think this is the right behavior and forcing text that comes from an LTR language like English to be displayed on the right was wrong. 
See the last point 
image
from https://m2.material.io/design/usability/bidirectionality.html#mirroring-layout

@seadowg do you agree?

**Comment by seadowg** [comment]
Not sure I understand the problem here. Either way, I reckon this would be better to discuss with @alyblenkin.

**Comment by grzesiek2010** [comment]
> Not sure I understand the problem here.

We used to force text that is from LTR languages to be displayed on the right side if the layout is RTL and the other way around. To me, this approach was wrong and also caused issues like https://github.com/getodk/collect/issues/5684

**Comment by alyblenkin** [comment]
Hey @grzesiek2010, sorry for the slow response on this one! I agree that it was wrong to display LTR language on the right with RTL layouts. So we should always default to the language direction: 

LTR language + RTL layout = displayed on the left 
RTL language + LTR layout = displayed on the right

**Comment by grzesiek2010** [comment]
Thanks @alyblenkin so I'm adding the `needs testing` label back.

**Comment by srujner** [comment]
@grzesiek2010 And what about differences between Android 5.1 and 13 (at the screenshots above)? Which one is the correct behavior?

**Comment by grzesiek2010** [comment]
13 is the correct one and that is what you should see on most of the Android versions. If there is something wrong with the oldest versions Android 5 or 5 and 6 that's acceptable I would say.

**Comment by dbemke** [comment]
Selects with columns, texts and images look like that with RTL in Collect settings and All widgets form - is it ok?  Users might find this case confusing - which picture is in A or B answer.
selectImages

**Comment by grzesiek2010** [comment]
To me, it's a correct behavior. It's in RTL so elements are displayed from right to left but the labels come from an LTR language so they are on the left (in their containers). I would say that using LTR labels with RTL layout doesn't make sense so that's why it's confusing. I hope our users won't do that.

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on device with Android 10

**Verified cases:**
- settings in the form from the issue #5684 work as expected
- RTL, LTR set in the device and in Collect
- RTL, LTR settings in a form with selects

**Comment by srujner** [comment]
**Tested with Success!**

Verified on device with Android 5.1 and 13

**Comment by alyblenkin** [comment]
It's a good point @dbemke, this would be very confusing for users. I think having a two-column grid on mobile also adds to the confusion here. Ideally, we stick to one column, so it's very clear what to select - I've added this to my [notes](https://docs.google.com/document/d/1SLmcjraGy2T_b3SchePN_von-akrbCIv3J6NDLSWDqI/edit) as something to explore.

**Pull request merged**